src/style/color/hsluv.rs

```diff
@@ -0,0 +1,110 @@
+use std::{cmp::Ordering, f64::consts::PI};
+
+const M: [[f64; 3]; 3] = [
+    [3.240969941904521, -1.537383177570093, -0.498610760293],
+    [-0.96924363628087, 1.87596750150772, 0.041555057407175],
+    [0.055630079696993, -0.20397695888897, 1.056971514242878],
+];
+
+const REF_Y: f64 = 1.0;
+const REF_U: f64 = 0.19783000664283;
+const REF_V: f64 = 0.46831999493879;
+// CIE LUV constants
+const KAPPA: f64 = 903.2962962;
+const EPSILON: f64 = 0.0088564516;
+
+#[allow(clippy::many_single_char_names)]
+pub(super) fn hsluv_to_rgb(h: f64, s: f64, l: f64) -> (f64, f64, f64) {
+    // Convert HSLuv to LCH
+    let (l, c, h) = match l {
+        l if l > 99.9999999 => (100.0, 0.0, h),
+        l if l < 0.00000001 => (0.0, 0.0, h),
+        _ => {
+            let mx = max_chroma_for(l, h);
+            let c = mx / 100.0 * s;
+            (l, c, h)
+        }
+    };
+
+    // Convert LCH to LUV
+    let h_radians = h * PI / 180.0;
+    let u = h_radians.cos() * c;
+    let v = h_radians.sin() * c;
+
+    // Convert LUV to XYZ
+    if l == 0.0 {
+        return (0.0, 0.0, 0.0);
+    }
+    let var_y = if l > 8.0 {
+        REF_Y * ((l + 16.0) / 116.0).powf(3.0)
+    } else {
+        REF_Y * l / KAPPA
+    };
+    let var_u = u / (13.0 * l) + REF_U;
+    let var_v = v / (13.0 * l) + REF_V;
+
+    let y = var_y * REF_Y;
+    let x = 0.0 - (9.0 * y * var_u) / ((var_u - 4.0) * var_v - var_u * var_v);
+    let z = (9.0 * y - (15.0 * var_v * y) - (var_v * x)) / (3.0 * var_v);
```

Comment:
Is this part panic free? Also, it would be nice to explain these little sections via inner comments for people who are not familiar with the conversion.

Reply:
yes, all the variables are `f64` and there won't be any division by 0

Reply:
I will add some comments